###**Anotaciones Semana 2 prob y estat**###


***Poblacion***: Es el conjunto de individuos que vamos a utilizar para nuestros analisis.

***P.Finita:*** **Tiene un fin aparente**

***P.Infinita:*** **no tiene fin**

***Muestras:***Es el conjunto o subconjunto representativo que es seleccionado para la población, que incluya todas las caracteristicas de la población.

***1-*** Debe ser **REPRESENTATIVA**

***2-***Tiene que tener un tamaño **DEMOSTRATIVO**

***3-***Tiene que ser **DIVERSO** los **DATOS**

***Paramétro:*** Valor numerico que describe una característica de la población.

***Valor estadístico:***Es lo **CONTRARIO** del **PARAMETRO**, es usado para asumir o determinar algo de la población.

***Muestreo:***Este tiene 2 tipos.

***Aleatorio o probabilistico:***tiene una igual probabilidad para la muestra.

***No probabilistico-SISTEMATICO:*** Un sistema aleatorio para elegir.

***Aleatorio simple:***Numeros al azar con programacion.

***Datos estartificados:***Dividir nuestro universo en conjuntos llamados estratos, seleccionados proporcionalmente.

***Por Conglomerados:*** Se seleccionan los grupos para haacer una muestra de una poblacion grande.

***Muestreo no probabilistica:*** muestras no aleatorias.

segmentacion por **CONVENIENCIA**

**MUESTREO** por **JUICIO**

***Muestreo por cuotas:***con proporciones especificas.

MUESTRA**(SNOW BALL)**

##configuracion del ambiente de trabajo

In [ ]:
import pandas as pd
data = pd.read_csv('/content/SoftwareProject_DataSet.csv')

#Verificacion del contenido del DataSet

In [ ]:
data.head(n = 10)

Proyecto    Módulo       Desarrollador  \
0      Smith, Torres and Flowers      long       Patricia Ford   
1                Jordan and Sons      lose      Melissa Briggs   
2     Madden, Jackson and Keller  thousand    Alexander Spence   
3                    Evans Group     table      Michael Porter   
4                    Pearson LLC     light      Andrew Gregory   
5     Williams, Mckee and Kelley  audience        Jane Donovan   
6  Harrison, Carroll and Barnett  attorney  Michael Fitzgerald   
7                     Bean-Payne    decide    Jessica Williams   
8           Fitzpatrick and Sons     small        Steven Allen   
9                     Ray-Arnold      term       Deborah Davis   

   Líneas de código  Número de bugs  Tiempo de desarrollo (horas)  \
0              4227              39                           375   
1              6124              37                           132   
2               588              30                           441   
3              5796              17                            87   
4              3019              11                           263   
5              6300              41                           353   
6              3932              36                           420   
7               514              28                           102   
8              4928               9                           341   
9              3378              14                           208   

   Costo (USD)  Satisfacción del cliente  Antigüedad del desarrollador (años)  \
0        84053                         1                                   10   
1        82167                         5                                    9   
2        75204                         3                                    7   
3        32147                         1                                    1   
4        44998                         2                                    5   
5        61827                         3                                   15   
6         6515                         2                                    9   
7        20075                         1                                    2   
8         4663                         2                                    5   
9        50665                         5                                    2   

  Idioma de programación Plataforma   Tipo de proyecto Fase del proyecto  \
0                   Java     Mobile        Application        Deployment   
1                 Kotlin     Mobile  Management System           Testing   
2             Javascript     Mobile        Application        Deployment   
3                 Kotlin     Mobile        Application            Design   
4                 Python    Desktop           Web site       Development   
5                 Kotlin     Mobile  Management System       Development   
6                 Kotlin     Mobile        Application           Testing   
7                 Kotlin     Mobile        Application        Deployment   
8                 Kotlin     Mobile           Web site        Deployment   
9                   Java    Desktop           Web site       Development   

   Tamaño del equipo                 Tecnologías  
0                  7   professor, evidence, down  
1                 10           clear, rich, tend  
2                 10        process, cut, debate  
3                  1     bank, management, those  
4                  3          you, visit, camera  
5                  4        among, together, eye  
6                  6       popular, almost, hour  
7                  4  Democrat, establish, house  
8                  2        down, consumer, song  
9                  1            fall, major, arm

###Obtener el tamañao de la poblacion

##creacion de una muestra aleatoria simple de 50 registros

In [ ]:
sample_size = 50
sample = data.sample(n= sample_size, random_state=12)

In [ ]:
population_size = len(data)
print("el tamaño de la poblacion es",population_size)

el tamaño de la poblacion es 10000


##Revision de la Muestra

In [ ]:
display(sample)

Proyecto        Módulo          Desarrollador  \
5669        Powell, Mcbride and Steele       product            James Woods   
8800                     Lozano-Garcia          just            Chad Hodges   
3205                      Powell-Smith         smile            James Poole   
8731            Hood, Mann and Delgado         class        Kathleen Jordan   
6412                     Barnes-Newman          talk         Edward Webster   
6828                     Maxwell-Smith      official             Scott Bell   
5795       Maxwell, Scott and Phillips            of          Linda Johnson   
9617                        Webb Group            PM             Robin Mann   
4643                   Castro and Sons         claim           Samuel Henry   
1844         Gonzales, Miller and Lamb       popular            Erica Boyer   
7947                     Horne-Wallace         exist              Mark Byrd   
6792       Jackson, Gilbert and Jordan       provide        Cheryl Robinson   
1237                  Edwards and Sons  relationship         Michael Greene   
5974  Gallagher, Sanchez and Velasquez       mention        Robert Shepherd   
4498                  Hawkins-Martinez         level            Diane Allen   
8624                 Chambers and Sons         again          Matthew Mills   
13          Armstrong, Hudson and Dunn       medical         Brandon Martin   
555                  Robinson and Sons      approach             Jorge King   
7787                        Kirk-Moore        public            Joan Foster   
4007       Calhoun, Willis and Collins       company              David Lee   
8015                    Norris-Manning         seven        Brandon Bonilla   
2890           Jones, Brown and Murphy        record    Stephanie Rodriguez   
8322                   Gonzales-Morris    discussion          Timothy Ochoa   
5665                  Valenzuela-Jones         agree       Gabriela Krueger   
2568                       Johnson-Lee        middle              Ian Lopez   
812       Mitchell, Mcmillan and White       concern       Jeffrey Martinez   
3064                        Vega Group         stock        Jeremy Richmond   
3721                        Garcia PLC           you           Matthew Hood   
190                     Nguyen-Stevens           ask           Lisa Harrell   
5623                        Harris PLC       message            Diana Garza   
4538                   Clements-Miller          cost         Joshua Mcguire   
2464                     Carlson-Terry     recognize         Joshua Hendrix   
302       Anderson, Murphy and Johnson        doctor          Lauren Lucero   
4124                         Davis PLC         class            Tammy White   
6382                  Marquez and Sons           for       Elizabeth Spears   
8242   Thornton, Washington and Wilson        nearly           Linda Warren   
9901                    Kirby and Sons        report          Joshua Jacobs   
3496                      Reilly-Smith   environment             Becky Boyd   
2585                          Hill Inc          also           Destiny Hale   
4170                    Chandler Group         skill            Anne Garcia   
9221                       Austin-Park      director          Pamela Benson   
4667         Garcia, Ortega and Howard       million  Christopher Dominguez   
9743          Taylor, Crosby and Smith      election            Faith Mason   
3083                        Ramsey LLC     professor           Justin Perez   
7816                  Griffin and Sons       finally            Walter Bell   
3005         Thomas, Reyes and Vincent          must         Vickie Sanders   
8700                  Bartlett-Collier          hand       Danielle Morales   
7896                   Sanchez-Johnson           yes          Ashley Murphy   
5875          Wilson, Huynh and Newton          fear           Ashley Brown   
6403                      Deleon-Moran    especially               Kim Rice   



In [ ]:
sample_size = len(sample)
print(sample_size)

50


##Muestreo Aleatorio Sistematico

##seleccion sistematica de la mauestra

##Definicion del intervalo k(tamaño de la poblacion / el tamaño de la muestra deseada)

In [ ]:
sample_size = 100
interval_k = population_size//sample_size

##seleccion de muestra sistematica

In [ ]:
systematic_sample = data.iloc[::interval_k]

In [ ]:
print("Muestra obtenida por muestreo sistematico")
display(systematic_sample)

Muestra obtenida por muestreo sistematico


Proyecto      Módulo        Desarrollador  \
0         Smith, Torres and Flowers        long        Patricia Ford   
100                   Bell and Sons      forget      Katie Carpenter   
200         Hughes, West and Austin     purpose      Kristina Palmer   
300                 Howard and Sons    thousand      Elizabeth Lopez   
400                    Lawrence PLC     surface          Paula Smith   
...                             ...         ...                  ...   
9500  Adams, Simmons and Richardson  understand      Richard Johnson   
9600                 Fuller-Johnson  Republican   Julia Matthews DDS   
9700                 Robinson Group    consider        Caleb Skinner   
9800        Lopez, Miller and Myers      growth  Christopher Salazar   
9900               Chapman-Phillips      detail          Jeremy Ruiz   

      Líneas de código  Número de bugs  Tiempo de desarrollo (horas)  \
0                 4227              39                           375   
100               5953               7                           251   
200               1425              41                           131   
300               4218               8                           223   
400               5279               9                           482   
...                ...             ...                           ...   
9500              6457              38                           334   
9600              6689              31                            67   
9700              5453              35                            80   
9800              3380              21                           398   
9900              7797              35                            82   

      Costo (USD)  Satisfacción del cliente  \
0           84053                         1   
100         48438                         5   
200         42944                         1   
300         33948                         1   
400         80195                         2   
...           ...                       ...   
9500        76598                         4   
9600        67997                         1   
9700        62104                         1   
9800        15510                         2   
9900        35103                         1   

      Antigüedad del desarrollador (años) Idioma de programación Plataforma  \
0                                      10                   Java     Mobile   
100                                     9             Javascript        Web   
200                                     2                 Kotlin     Mobile   
300                                    10                 Python     Mobile   
400                                    15                 Kotlin    Desktop   
...                                   ...                    ...        ...   
9500                                    3                 Python    Desktop   
9600                                   14             Javascript        Web   
9700                                   13                   Java    Desktop   
9800                                    3                 Kotlin     Mobile   
9900                                    5                   Java        Web   

       Tipo de proyecto Fase del proyecto  Tamaño del equipo  \
0           Application        Deployment                  7   
100         Application           Testing                 10   
200   Management System           Testing                  2   
300   Management System           Testing                  3   
400            Web site            Design                  1   
...                 ...               ...                ...   
9500  Management System       Development                  2   
9600  Management System        Deployment                  3   
9700        Application            Design                  1   
9800        Application           Testing                  1   
9900  Management System        Deployment                  7   

              

### Agrupacion de datos y subpoblaciones

## dividir la poblacion por un factor

In [ ]:
grouped_by_dev_language = data.groupby("Idioma de programación")
grouped_by_dev_language.head()

Proyecto       Módulo          Desarrollador  \
0       Smith, Torres and Flowers         long          Patricia Ford   
1                 Jordan and Sons         lose         Melissa Briggs   
2      Madden, Jackson and Keller     thousand       Alexander Spence   
3                     Evans Group        table         Michael Porter   
4                     Pearson LLC        light         Andrew Gregory   
5      Williams, Mckee and Kelley     audience           Jane Donovan   
6   Harrison, Carroll and Barnett     attorney     Michael Fitzgerald   
7                      Bean-Payne       decide       Jessica Williams   
9                      Ray-Arnold         term          Deborah Davis   
10                     Palmer PLC          age        Zachary Johnson   
12                Valentine-Myers       ground  Mr. Alvin Gardner DVM   
13     Armstrong, Hudson and Dunn      medical         Brandon Martin   
15                Barnes and Sons     remember          Joshua Mosley   
16                     Watson Inc         kind          Scott Mcmahon   
17                      Allen Ltd  performance            Renee Chung   
18               Simmons and Sons          end     Kathleen Schneider   
19                   Duncan Group          now         Jennifer Berry   
22                 Payne and Sons          tax          Trevor Gibson   
26                      Perez Ltd       father      Pamela Villarreal   
28               Garcia-Valentine        guess          Heather Adams   

    Líneas de código  Número de bugs  Tiempo de desarrollo (horas)  \
0               4227              39                           375   
1               6124              37                           132   
2                588              30                           441   
3               5796              17                            87   
4               3019              11                           263   
5               6300              41                           353   
6               3932              36                           420   
7                514              28                           102   
9               3378              14                           208   
10              5477              38                           142   
12              2040              47                           396   
13              6390              13                           178   
15              3330               8                            82   
16              1487              36                           110   
17              3057              21                           176   
18              5277               4                           428   
19              1074              33                           473   
22              7553               9                           201   
26              4811              36                           281   
28              2565              27                           432   

    Costo (USD)  Satisfacción del cliente  \
0         84053                         1   
1         82167                         5   
2         75204                         3   
3         32147                         1   
4         44998                         2   
5         61827                         3   
6          6515                         2   
7         20075                         1   
9         50665                         5   
10        72981                         1   
12        28246                         1   
13         9107                         1   
15        33327                         5   
16        49796                         5   
17        70016                         3   
18        27642                         4   
19        79586                         2   
22        95320                         5   
26        28183                         2   
28        94713                         4   

    Antigüedad del desarrollador (años) Idioma de programación Plataforma 

In [ ]:
group_analysis = grouped_by_dev_language[["Líneas de código","Número de bugs"]].mean()
print("analisis por Tamaño de equipo")
display(group_analysis)

analisis por Tamaño de equipo


Líneas de código  Número de bugs
Idioma de programación                                  
Java                         5046.673913       24.454193
Javascript                   5038.137720       25.222722
Kotlin                       5018.528842       25.106091
Python                       5128.660528       25.423139

##Analisis por tiempo y numero del equipo.


In [ ]:
group_analysis = grouped_by_dev_language[["Tiempo de desarrollo (horas)","Tamaño del equipo"]].mean()
print("analisis por lenguaje de programacion")
display(group_analysis)

analisis por lenguaje de programacion


Tiempo de desarrollo (horas)  Tamaño del equipo
Idioma de programación                                                 
Java                                      250.220885           5.532609
Javascript                                255.118104           5.490396
Kotlin                                    251.197660           5.488503
Python                                    262.393515           5.462770

###Analisis del caso###


              Tiempo              Tamaño del equipo      Promedio
Java-      250.22 /////////////               5.53  ///////////// 45.24

Javascript-  255.11 /////// 5.49 ////////// 46.46

Kotlin- 251.19 ///////////// 5.48 ////////// 45.83

Python- 262.39 ////////// 5.46 ///////// 48.05

###Conclusión###
**Mientras haya mas personas trabajando en un equipo, menor es el tiempo para terminar el proyecto:**

****Yo elegiría a a Java, ya que tiene el tamaño de equipo mas grande, un tiempo menor al de los otros equipos, y en si su promedio es menor.****  

###Analisis por Líneas de código y Tiempo de desarrollo (horas)

###***CONCLUSIÓN***###
**El análisis muestra que Kotlin es el lenguaje más eficiente en términos de líneas de código, mientras que Java requiere menos tiempo de desarrollo. Por otro lado, Python, aunque es conocido por su simplicidad, genera más líneas y requiere más tiempo, posiblemente debido a la naturaleza de los proyectos en los que se utiliza. La elección del lenguaje depende del equilibrio entre tiempo, líneas de código y las necesidades específicas del proyecto.**

###**Objetivo**###
***Analizar y comparar el tiempo de desarrollo (horas) y las líneas de código requeridas por desarrolladores utilizando diferentes lenguajes de programación (Java, JavaScript, Kotlin y Python), con el propósito de identificar patrones de eficiencia y proporcionar información para la toma de decisiones en futuros proyectos de software.***

In [ ]:
grouped_by_line_code = data.groupby("Idioma de programación")
grouped_by_line_code.head()


Proyecto       Módulo          Desarrollador  \
0       Smith, Torres and Flowers         long          Patricia Ford   
1                 Jordan and Sons         lose         Melissa Briggs   
2      Madden, Jackson and Keller     thousand       Alexander Spence   
3                     Evans Group        table         Michael Porter   
4                     Pearson LLC        light         Andrew Gregory   
5      Williams, Mckee and Kelley     audience           Jane Donovan   
6   Harrison, Carroll and Barnett     attorney     Michael Fitzgerald   
7                      Bean-Payne       decide       Jessica Williams   
9                      Ray-Arnold         term          Deborah Davis   
10                     Palmer PLC          age        Zachary Johnson   
12                Valentine-Myers       ground  Mr. Alvin Gardner DVM   
13     Armstrong, Hudson and Dunn      medical         Brandon Martin   
15                Barnes and Sons     remember          Joshua Mosley   
16                     Watson Inc         kind          Scott Mcmahon   
17                      Allen Ltd  performance            Renee Chung   
18               Simmons and Sons          end     Kathleen Schneider   
19                   Duncan Group          now         Jennifer Berry   
22                 Payne and Sons          tax          Trevor Gibson   
26                      Perez Ltd       father      Pamela Villarreal   
28               Garcia-Valentine        guess          Heather Adams   

    Líneas de código  Número de bugs  Tiempo de desarrollo (horas)  \
0               4227              39                           375   
1               6124              37                           132   
2                588              30                           441   
3               5796              17                            87   
4               3019              11                           263   
5               6300              41                           353   
6               3932              36                           420   
7                514              28                           102   
9               3378              14                           208   
10              5477              38                           142   
12              2040              47                           396   
13              6390              13                           178   
15              3330               8                            82   
16              1487              36                           110   
17              3057              21                           176   
18              5277               4                           428   
19              1074              33                           473   
22              7553               9                           201   
26              4811              36                           281   
28              2565              27                           432   

    Costo (USD)  Satisfacción del cliente  \
0         84053                         1   
1         82167                         5   
2         75204                         3   
3         32147                         1   
4         44998                         2   
5         61827                         3   
6          6515                         2   
7         20075                         1   
9         50665                         5   
10        72981                         1   
12        28246                         1   
13         9107                         1   
15        33327                         5   
16        49796                         5   
17        70016                         3   
18        27642                         4   
19        79586                         2   
22        95320                         5   
26        28183                         2   
28        94713                         4   

    Antigüedad del desarrollador (años) Idioma de programación Plataforma 

In [ ]:
group_analysis = grouped_by_line_code[["Líneas de código", "Tiempo de desarrollo (horas)"]].mean()
print("Analisis por Líneas de código")
display(group_analysis)

Analisis por Líneas de código


Líneas de código  Tiempo de desarrollo (horas)
Idioma de programación                                                
Java                         5046.673913                    250.220885
Javascript                   5038.137720                    255.118104
Kotlin                       5018.528842                    251.197660
Python                       5128.660528                    262.393515

##Creación de una muestra estratificada por "Tipos de proyectos"




##Apuntes:##
Los programadores tienen una convención la cual dice que el máximo de carácteres en una línea de código debe ser 100, para tener un orden en el código y que sea legible para todos.
Si un código es muy largo, debemos hacer indentación al código. La indentación se realiza automáticamente al hacer TAB estando en la siguiente línea de código.

# Asignación: #
Investigar cómo son y cómo se generan los historigramas, diagramas de tallo y hoja y diagramas de caja, de preferencia ejemplos.

In [ ]:
grouped_by_project_type = data.groupby("Tipo de proyecto")


In [ ]:
stratified_sample = grouped_by_project_type.sample(n=100, random_state=42)
print("Muestra estratificada")
display(stratified_sample)

Muestra estratificada


Proyecto    Módulo      Desarrollador  \
3275                    Green Inc      ever        Erik Finley   
6914                 Smith-Miller  approach     Tony Carpenter   
2315           Hamilton-Gutierrez    window    Jacqueline Neal   
8422                  Stewart LLC      hair  Michael Hernandez   
546   Brewer, Sanders and Wilkins       see       Devon Flores   
...                           ...       ...                ...   
802                 Mata-Peterson    health      Chris Salazar   
1340    Mejia, Carrillo and Brown      than     Janet Ferguson   
3320                    Gomez PLC     young      Ryan Mcdonald   
2545  Johnson, Peters and Robbins   society    Jennifer Wright   
9474                Johnson Group     today       Brian Cooper   

      Líneas de código  Número de bugs  Tiempo de desarrollo (horas)  \
3275              5753              42                            36   
6914               254              39                            70   
2315              7342              12                           346   
8422              4961              16                           317   
546               4198              49                           144   
...                ...             ...                           ...   
802               5980               1                           436   
1340              4943              36                           305   
3320              2848              21                           343   
2545              8757              28                           133   
9474              1162              38                           475   

      Costo (USD)  Satisfacción del cliente  \
3275        19997                         5   
6914        44964                         3   
2315        40812                         1   
8422        57670                         3   
546         18838                         3   
...           ...                       ...   
802         95856                         2   
1340        59785                         5   
3320        71963                         4   
2545        37717                         2   
9474        13357                         5   

      Antigüedad del desarrollador (años) Idioma de programación Plataforma  \
3275                                    3                   Java        Web   
6914                                    6                   Java     Mobile   
2315                                    8             Javascript        Web   
8422                                    6                 Python    Desktop   
546                                     4                 Kotlin     Mobile   
...                                   ...                    ...        ...   
802                                     6                 Python        Web   
1340                                    9             Javascript     Mobile   
3320                                    6                 Kotlin     Mobile   
2545                                    8                   Java    Desktop   
9474                                   15                 Python        Web   

     Tipo de proyecto Fase del proyecto  Tamaño del equipo  \
3275      Application            Design                 10   
6914      Application       Development                  1   
2315      Application           Testing                  6   
8422      Application            Design                  8   
546       Application            Design                  1   
...               ...               ...                ...   
802          Web site       Development                  3   
1340         Web site           Testing                  3   
3320         Web site           Testing                  2   
2545         Web site           Testing                  7   
9474         Web site           Testing                 10   

                  Tecnologías  
3275      trial, memory, soon  
6914  lose, practice, similar  
2315       d

##Muestreo por conveniencia

In [ ]:
convenience_sample = data.head(50)
print("Sample of the first row of the DataFrame")
display(convenience_sample)

Sample of the first row of the DataFrame


Proyecto       Módulo          Desarrollador  \
0       Smith, Torres and Flowers         long          Patricia Ford   
1                 Jordan and Sons         lose         Melissa Briggs   
2      Madden, Jackson and Keller     thousand       Alexander Spence   
3                     Evans Group        table         Michael Porter   
4                     Pearson LLC        light         Andrew Gregory   
5      Williams, Mckee and Kelley     audience           Jane Donovan   
6   Harrison, Carroll and Barnett     attorney     Michael Fitzgerald   
7                      Bean-Payne       decide       Jessica Williams   
8            Fitzpatrick and Sons        small           Steven Allen   
9                      Ray-Arnold         term          Deborah Davis   
10                     Palmer PLC          age        Zachary Johnson   
11                   Anderson Inc         girl     Vanessa Richardson   
12                Valentine-Myers       ground  Mr. Alvin Gardner DVM   
13     Armstrong, Hudson and Dunn      medical         Brandon Martin   
14                 Smith and Sons        never          Douglas Moore   
15                Barnes and Sons     remember          Joshua Mosley   
16                     Watson Inc         kind          Scott Mcmahon   
17                      Allen Ltd  performance            Renee Chung   
18               Simmons and Sons          end     Kathleen Schneider   
19                   Duncan Group          now         Jennifer Berry   
20                    Vaughan Inc     describe          Joseph Sutton   
21                    Moore Group         vote           Ashley Smith   
22                 Payne and Sons          tax          Trevor Gibson   
23                 Jackson-Palmer         west        James Underwood   
24                 Collier-Hunter         news       Charles Gonzalez   
25     Morales, Porter and Wilson          but        Scott Cervantes   
26                      Perez Ltd       father      Pamela Villarreal   
27      Jackson, Le and Hernandez         fact         Jessica Ibarra   
28               Garcia-Valentine        guess          Heather Adams   
29                     Moore-Case       player             Luis Rojas   
30     Crawford, Hart and Roberts  performance           Shane Rivera   
31                 Fitzgerald Ltd        major          Madison Munoz   
32                  Krueger Group          son           Doris Lozano   
33     Taylor, Alvarado and Hayes        reach          Alexis Palmer   
34       Sanchez, Ball and Foster         born       Christopher Rice   
35                    Wong-Torres      purpose      Amanda Coleman MD   
36             Franklin-Hernandez         also         Matthew Fuller   
37                 Hill-Underwood       forget       Melissa Harrison   
38    Bell, Gillespie and Spencer        model         Cynthia Nelson   
39                    Jones-Chase        trial          Haley Griffin   
40                    Mendoza Ltd          red           Melissa Haas   
41        Wilson, Le and Williams         cell        Jonathon Turner   
42                     Montes-Day     military              John Levy   
43                   Briggs-Green         thus          Jennifer Diaz   
44                     Moore-Mack         deal         Sandra Holland   
45     Williams, Adams and Martin         hour          Melissa Doyle   
46                   Bryant-Black          bit             Juan Young   
47      Palmer, Owens and Farrell         poor          Jamie Aguilar   
48            Mcdaniel-Montgomery      million         Arthur Jackson   
49       Nielsen, Smith and Smith        catch            John Martin   

    Líneas de código  Número de bugs  Tiempo de desarrollo (horas)  \
0               4227              39                           375   
1               6124              37                           132   
2                588              30                           441   
3               5796 

##Comparación en los métodos de muestreo

In [ ]:
methods_comparison = pd.DataFrame({
"Población General": data["Tiempo de desarrollo (horas)"].mean(),
"Aleatorio Simple": sample["Tiempo de desarrollo (horas)"].mean(),
"Aleatorio Sistemático": systematic_sample["Tiempo de desarrollo (horas)"].mean(),
"Aleatoria Estratificada": stratified_sample["Tiempo de desarrollo (horas)"].mean(),
"No Aleatorio por conveniencia": convenience_sample["Tiempo de desarrollo (horas)"].mean(),
}, index=["Media de Horas de Desarrollo"])
print("Comparacion de Metodos");
display(methods_comparison)

Comparacion de Metodos


Población General  Aleatorio Simple  \
Media de Horas de Desarrollo           254.7021            269.32   

                              Aleatorio Sistemático  Aleatoria Estratificada  \
Media de Horas de Desarrollo                 257.51                    258.0   

                              No Aleatorio por conveniencia  
Media de Horas de Desarrollo                         267.82

##Muestra estratificada por lenguaje


In [ ]:
grouped_by_language_sample = data.groupby("Idioma de programación")

In [ ]:
stratified_sample = grouped_by_language_sample.sample(n=5, random_state=40)
print("Muestra estratificada por lenguaje")
stratified_sample_sorted = stratified_sample.sort_values("Idioma de programación")
display(stratified_sample_sorted)

Muestra estratificada por lenguaje


Proyecto   Módulo      Desarrollador  \
7784         Boone, Pace and Harvey  economy      Jeffrey Jones   
8836        West, Pollard and Riley   myself         Diane Todd   
9892                Cherry and Sons   forget        Lori Martin   
9852    Hoffman, White and Gonzalez     pass       Brian Pierce   
3054                      Moore Ltd     upon      Jeremy Miller   
3509     Powell, Evans and Espinoza     pass       Toni Johnson   
1661                     Perry-Mack     thus       Thomas Young   
2161                   Wilson-Avery    laugh         Anna Young   
1509                  Silva-Bridges     film   Charles Odonnell   
1489  Frazier, Lambert and Mitchell   single      Harold Haynes   
6454             Anderson-Rodriguez   almost      David Johnson   
2842              Guzman-Cunningham  station     Courtney Moody   
4177                  Young-Johnson  reflect       Tanya Harper   
5932                  King and Sons     body      Dillon Rhodes   
5298     Dyer, Escobar and Mckenzie    hotel     Ashley Roberts   
699                       Clark Ltd    point   Andrea Henderson   
3129      Kelley, Barnes and Miller   animal      James Johnson   
2969                 Clark and Sons       or    Jonathan Franco   
4919        Jones, Valdez and Woods     care     James Figueroa   
3894                    Cline-Brown     fill  Bradley Hernandez   

      Líneas de código  Número de bugs  Tiempo de desarrollo (horas)  \
7784              6191              38                           273   
8836              1073              21                           186   
9892              8605              50                            31   
9852              8058              38                           341   
3054              2017              50                           134   
3509              9301               1                           247   
1661              5184               4                           329   
2161              4972              23                           178   
1509              6934              32                           185   
1489              9593              17                           316   
6454              5118              36                           215   
2842              1736              10                            36   
4177              1343              26                            23   
5932              6958              38                            77   
5298              9611              15                           375   
699               3118               3                           448   
3129              3922              43                           440   
2969              9098              35                            54   
4919              2605              13                           214   
3894              8725              29                           391   

      Costo (USD)  Satisfacción del cliente  \
7784         8962                         5   
8836         1149                         4   
9892        31562                         3   
9852        34831                         3   
3054        94828                         5   
3509        35637                         4   
1661        77981                         2   
2161        21477                         5   
1509        14730                         1   
1489        50010                         1   
6454        61536                         3   
2842        76146                         3   
4177        59024                         1   
5932        42421                         1   
5298        22902                         5   
699         17470                         3   
3129        27618                         4   
2969         7918                         5   
4919        15557                         2   
3894        46540                         2   

      Antigüedad del desarrollador (años) Idioma de programación Plataforma  \
7784                                   